### PR TITLE
fix(webhook): add a queue for duplicate webhooks so that they be reactived later

### PR DESCRIPTION
**Pull request by johnBgood**
## Description

ℹ️ I plan to backport this PR to 8.7. Let me know if 8.6 makes sense as well. 

- This pull request introduces a queueing mechanism for webhook connector registration, allowing multiple connectors to wait for activation on the same webhook path and improving the handling of conflicting registrations.
- It also refactors logging to reduce noise by lowering log levels from debug to trace in several classes.
- Additionally, new tests are added to ensure the correct behavior of the webhook waiting queue. The most important changes are as follows:

**Webhook connector registration and queueing enhancements:**

* Added the new `WebhookWaitingQueue` class to manage a queue of connectors waiting to register on the same webhook context path, with logic to handle queue limits, health status, and activation sequencing. (`WebhookWaitingQueue.java`)
* Updated `WebhookConnectorRegistry` to use the `WebhookWaitingQueue` for managing connector registration conflicts, replacing the previous immediate error with queueing and standby logic. Now, when a connector is deregistered, the next connector in the queue (if any) is activated. (`WebhookConnectorRegistry.java`) [[1]](diffhunk://#diff-006e48f0ecf463e2725bd520952cf980be96e2150b02923c057749c2217fe91bR32-R33) [[2]](diffhunk://#diff-006e48f0ecf463e2725bd520952cf980be96e2150b02923c057749c2217fe91bL54-R57) [[3]](diffhunk://#diff-006e48f0ecf463e2725bd520952cf980be96e2150b02923c057749c2217fe91bR81-R93)
* Changed the activation order in `BatchExecutableProcessor` to ensure the connector is activated before being registered in the webhook registry. (`BatchExecutableProcessor.java`)

**Testing improvements:**

* Added a comprehensive test suite for the new `WebhookWaitingQueue` to verify queue behavior, health status updates, queue limits, and multi-context support. (`WebhookWaitingQueueTest.java`)
* Renamed and updated test files to reflect the new registry-based logic, e.g., `WebhookControllerPlainJavaTests.java` to `WebhookConnectorRegistryTest.java` and `WebhookControllerLogTest.java` to `WebhookConnectorRegistryLogTest.java`. (`WebhookConnectorRegistryTest.java`, `WebhookConnectorRegistryLogTest.java`) [[1]](diffhunk://#diff-83af8ff378303071ba0d2763ea60a3c40f7da67c02aa85fb96e794775d5f2d9dR31) [[2]](diffhunk://#diff-fe400d98c541366051fbace64b49e00c391e26948a744283e697c4295dfd78e3L19-R20) [[3]](diffhunk://#diff-fe400d98c541366051fbace64b49e00c391e26948a744283e697c4295dfd78e3L43-R43)

**Logging adjustments:**

* Lowered several log statements from debug to trace level in process definition import and state store classes to reduce log verbosity. (`ProcessDefinitionImporter.java`, `ProcessDefinitionSearch.java`, `ProcessStateStoreImpl.java`) [[1]](diffhunk://#diff-33d76824e77e34da0177d8714bace363026d88bce91bf6b3690cadf7105c5c27L67-R67) [[2]](diffhunk://#diff-33d76824e77e34da0177d8714bace363026d88bce91bf6b3690cadf7105c5c27L81-R81) [[3]](diffhunk://#diff-ded0477205912b172054b4085b3b93b08520dd34383e980518106dc2e6760d02L57-R57) [[4]](diffhunk://#diff-78a725f61ca4c46214a248723cf331f6de1b91baf6c8fff5c530a566847cb054L50-R56) [[5]](diffhunk://#diff-78a725f61ca4c46214a248723cf331f6de1b91baf6c8fff5c530a566847cb054L66-R66)

These changes collectively improve the robustness and observability of webhook connector registration and activation, especially in scenarios with conflicting registrations.

## Related issues

closes https://github.com/camunda/connectors/issues/5133

## Checklist

- [x] PR has a **milestone** or the `no milestone` label.
- [ ] Backport labels are added if these code changes should be backported. No backport label is added to the latest release, as this branch will be rebased onto main before the next release.

### CODE REVIEW

File: connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/webhook/WebhookWaitingQueue.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+                .message(
+                    "Path \"" + context + "\" is now available. Connector has been activated."));
+    next.context().reportHealth(Health.up());
+  }
+
+  private boolean hasWaitingConnectors(Queue<RegisteredExecutable.Activated> queue) {
+    return queue != null && !queue.isEmpty();
+  }
+
+  public Map<String, Queue<RegisteredExecutable.Activated>> getWaitingConnectors() {
```
Comment (johnBgood) [inline connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/webhook/WebhookWaitingQueue.java:142]: we might want to print the map content every 60s ?

File: connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/webhook/WebhookExecutables.java [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+package io.camunda.connector.runtime.inbound.webhook;
+
+import io.camunda.connector.api.inbound.Activity;
+import io.camunda.connector.api.inbound.Health;
+import io.camunda.connector.api.inbound.InboundConnectorDefinition;
+import io.camunda.connector.api.inbound.Severity;
+import io.camunda.connector.runtime.inbound.executable.RegisteredExecutable;
+import java.util.*;
+
+public class WebhookExecutables {
```
Comment (sbuettner) [inline connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/webhook/WebhookExecutables.java:26]: `WebhookExecutablesByPath`?

File: connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/webhook/WebhookExecutables.java [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+  private static final String TAG_QUEUEING = "Queueing";
+  private static final String TAG_ACTIVATION = "Activation";
+
+  private final List<RegisteredExecutable.Activated> executables = new ArrayList<>();
+  private final String context;
```
Comment (sbuettner) [inline connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/webhook/WebhookExecutables.java:32]: What about having a single field for the executable that is currently active? Could simplify the code and list filtering.

File: connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/webhook/WebhookConnectorRegistry.java [hunk 4]
```diff
@@ -18,91 +18,61 @@
...
-              + context
-              + " already in use by: "
-              + elementIdsByProcessId.entrySet().stream()
-                  .map(e -> "process " + e.getKey() + "(" + e.getValue() + ")")
-                  .reduce((a, b) -> a + ", " + b)
-                  .orElse("");
-      LOG.debug(logMessage);
-      throw new RuntimeException(logMessage);
-    }
+    Optional.ofNullable(
```
Comment (sbuettner) [inline connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/webhook/WebhookConnectorRegistry.java:44]: Why should this be nullable if we create it when not present?

File: connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/webhook/WebhookConnectorRegistry.java [hunk 5]
```diff
@@ -18,91 +18,61 @@
...
 
   public void register(RegisteredExecutable.Activated connector) {
-    var properties = connector.context().bindProperties(CommonWebhookProperties.class);
-    var context = properties.getContext();
-    if (context == null) {
-      var logMessage = "Webhook path not provided";
-      LOG.debug(logMessage);
-      throw new RuntimeException(logMessage);
-    }
+    var context = getContext(connector);
```
Comment (sbuettner) [inline connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/webhook/WebhookConnectorRegistry.java:41]: What if this is null?

File: connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/executable/BatchExecutableProcessor.java [hunk 6]
```diff
@@ -171,11 +171,11 @@ private RegisteredExecutable activateSingle(
       return new ConnectorNotRegistered(validData);
     }
     try {
+      executable.activate(context);
```
[the remark this answers is not in the document]
Comment (johnBgood) [reply connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/executable/BatchExecutableProcessor.java:174]: Similar to other connectors, the `activate` is just = start the infrastructure, and the rest will be started later (e.g. Kafka consumer). 
I can also bring back the old code here if necessary

File: connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/webhook/WebhookExecutables.java [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+    downExecutables.remove(foundExecutable);
+    return true;
+  }
+
+  private boolean sameExecutables(
+      RegisteredExecutable.Activated executableA, RegisteredExecutable.Activated executableB) {
+    return executableA != null
+        && executableB != null
+        && executableA.context().getDefinition().equals(executableB.context().getDefinition());
+  }
```
[the remark this answers is not in the document]
Comment (johnBgood) [reply connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/webhook/WebhookExecutables.java:91]: fixed

File: connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/webhook/WebhookExecutables.java [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+    markActiveExecutableAsUp();
+    updateDownExecutablesErrorMessage();
+
+    return true;
+  }
+
+  public RegisteredExecutable.Activated getActiveWebhook() {
+    if (activeExecutable == null) {
+      throw new IllegalStateException(
+          "No active (health status = UP) webhooks found for the context: " + context);
```
[the remark this answers is not in the document]
Comment (johnBgood) [reply connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/webhook/WebhookExecutables.java:114]: I'll add a test but this should result in a 500 yes, which makes sense given the exception nature

File: connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/executable/BatchExecutableProcessor.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (johnBgood) [reply connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/executable/BatchExecutableProcessor.java:174]: I'll revert the change I don't need really it

**Review by sbuettner — APPROVED** [review]
Great work!

**Comment by johnBgood** [comment]
# QA information
## Test Environment
- Test required in (you can check both):
  - [ ] SaaS
  - [ ] SM
- [ ] Create a new test for this feature in our [Pre Release Tests](https://modeler.ultrawombat.com/projects/017779a8-9459-4feb-8b47-eeef8b2717ef--connectors-pre-release-test) for @Szik to be able to test

## Test Scope
*Please describe the test scope, happy path, edge cases that come to your mind, and whatever you think might relevant to test*

## Test Data
*Please provide the test data, if needed (files, URLs, code snippets, FEEL expressions)*
